viz/CMakeLists.txt

```diff
@@ -37,8 +40,11 @@ rock_vizkit_plugin(base-viz
         SonarVisualization.hpp
         PointcloudVisualization.hpp
         DepthMapVisualization.hpp
+        RigidBodyStateSE3Visualization.hpp
+        WrenchModel.hpp
+        WrenchVisualization.hpp
         ${OPTIONAL_HPP}
     DEPS base-types
     LIBS ${Boost_SYSTEM_LIBRARY}
-    DEPS_PKGCONFIG base-logging
+    DEPS_PKGCONFIG base-logging PrimitivesFactory osgViz
```

Comment:
I really believe we should consider other alternatives before adding `osgViz` as a "hard" dependency in `base/types`.

This is also breaking rock-core's build. 

@doudou

Reply:
In practice, base/types already depends on `gui/vizkit3d` which depends on osgViz. The (optional) dependency should definitely be explicitly added to the manifest, though.

Reply:
Yes, that will be a problem in constrained environments (i.e embedded computers running iodrivers_base based drivers). Can we please don't do that?

Reply:
The keyword was **optional** ;-). The dependency on osgViz that this PR adds is also optional as the whole `viz/` folder is disabled by the Rock CMake macros if vizkit3d is not available or if [ROCK_VIZ_ENABLED](https://github.com/rock-core/base-cmake/blob/master/modules/Rock.cmake#L217) is set to OFF.

base/types already depends **optionally** on `gui/vizkit3d`. In constrained environments, you usually exclude `gui/.*` which removes that particular dependency

Reply:
> the whole viz/ folder is disabled by the Rock CMake macros if vizkit3d is not available or if ROCK_VIZ_ENABLED is set to OFF

I missed that, sorry. I guess I was misled by the breaking but (which is probably breaking for some other reason I didn't look into). Did this pass on your CI?

Reply:
```
-- Checking for module 'PrimitivesFactory'
--   No package 'PrimitivesFactory' found
CMake Error at /usr/share/cmake-3.10/Modules/FindPkgConfig.cmake:415 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPkgConfig.cmake:593 (_pkg_check_modules_internal)
  /buildbot/squidbot-build/build/install/base/cmake/share/rock/cmake/Rock.cmake:288 (pkg_check_modules)
  /buildbot/squidbot-build/build/install/base/cmake/share/rock/cmake/Rock.cmake:383 (rock_find_pkgconfig)
  /buildbot/squidbot-build/build/install/base/cmake/share/rock/cmake/Rock.cmake:649 (rock_target_definition)
  /buildbot/squidbot-build/build/install/base/cmake/share/rock/cmake/Rock.cmake:811 (rock_library_common)
  viz/CMakeLists.txt:11 (rock_vizkit_plugin)
```

Reply:
I don't know what PrimitivesFactory is or where it comes from. I was looking at osgViz since it was what you mentioned in your comment.

Didn't check CI yet.

Reply:
PrimitivesFactory is a module in osgViz